pkg/sidecarexec/sandbox_cmd_runner_linux.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package sidecarexec
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	goexec "os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"syscall"
+
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/exec"
+)
+
+var (
+	limitDir = os.Getenv("KAPPCTRL_SIDECAREXEC_LIMIT_DIR")
+)
+
+// Run executes exec.Cmd under new process/mount namespace.
+func (r SandboxCmdRunner) Run(cmd *goexec.Cmd, opts exec.RunOpts) error {
+	newRootDir, err := ioutil.TempDir("", "sidecarexecwrap-cmd")
+	if err != nil {
+		return fmt.Errorf("Creating new root dir: %s", err)
+	}
+
+	wrapArgs := sandboxWrapArgs{
+		NewRootDir: newRootDir,
+		Posix:      r.opts.RequiresPosix != nil && r.opts.RequiresPosix[filepath.Base(cmd.Path)],
+		Network:    r.opts.RequiresNetwork != nil && r.opts.RequiresNetwork[filepath.Base(cmd.Path)],
+
+		CmdDir:  cmd.Dir,
+		CmdPath: cmd.Path,
+		CmdArgs: cmd.Args[1:], // drop first arg (binary name)
+
+		VisiblePaths: opts.VisiblePaths,
+	}
+
+	// Roughly equivalent to: unshare -Urm -R / ...
+	cmd.Path = "/usr/bin/kapp-controller"
+	cmd.Args = []string{"kapp-controller", "--sidecarexecwrap", wrapArgs.AsString()}
+	cmd.Dir = "/"
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		// TODO syscall.CLONE_NEWUTS | syscall.CLONE_NEWIPC | syscall.CLONE_NEWPID?
```

Comment:
agree with @neil-hickey that this TODO looks like it's not meant to be merged?   maybe you're planning to resolve it through testing in different iaas/infra/providers?

Reply:
low level linux stuff here that's hurting my head because I don't know enough about it all, but *IF* someone were TODO `syscall.CLONE_NEWPID` (creating a process ns for sidecarexecwrap process to run in) would that play nicely with the sidecarexecserver `reapZombies`? 

Should we `s/TODO/DONTDO/`

Reply:
CLONE_NEWPID might actually nicely to eliminate reaping of zombies since all procs (zombies or not) would be collected upon exit of the root of the pid namespace.

Reply:
ah! nice! if we can let the OS seamlessly handle cleaning up resources i'm in favor of that.

I don't think it's a blocker for this PR. But leaving the TODO's here as-is I think as time goes on, is less likely to be done.